lib/utils/templatize.js

```diff
@@ -121,11 +121,11 @@ export function showHideChildren(hide, children) {
       } else if (n.localName === 'slot') {
         if (hide) {
           n.__polymerReplaced__ = document.createComment('hidden-slot');
-          wrap(wrap(n).parentNode).replaceChild(n.__polymerReplaced__, n);
+          wrap(wrapIfNeeded(n).parentNode).replaceChild(n.__polymerReplaced__, n);
         } else {
           const replace = n.__polymerReplaced__;
           if (replace) {
-            wrap(wrap(replace).parentNode).replaceChild(n, replace);
+            wrap(wrapIfNeeded(replace).parentNode).replaceChild(n, replace);
```

Comment:
Not totally clear why the outer `wrap` can't be `wrapIfNeeded`.  Adding a `<slot>` to any random spot in a shadow root obviously wouldn't be automatically on-demand patched, but this is only putting a `<slot>` _back_ into a node that already had it (which I think means it is guaranteed to be patched)?

Reply:
Done.